### PR TITLE
Feature/recline alternatives

### PR DESCRIPTION
**Pull request by Bachibouzouk**
Merging roadmap : 

- Merge #499 first, then merge this PR

### CODE REVIEW

**Comment by Bachibouzouk** [comment]
@MGlauer - what is still required before we can merge this work into `develop` ?

**Comment by MGlauer** [comment]
> @MGlauer - what is still required before we can merge this work into `develop` ?

Essentially #491

**Comment by Bachibouzouk** [comment]
> Essentially #491

For this I needed to understand how the views.py worked in `dataedit` and I added docstrings to the functions in #497. Can't we merge this PR and then I implement the graph in a subsequent PR?

**Comment by MGlauer** [comment]
This would lead to us loosing functionality in the dev branch. I would like to avoid that. Also: The views are (for now) not part of this ticket. Tables and maps draw their data through the API and so should graphs.
The only thing, where a server-side handling would come into play, is when we add the view functionality. I would argue that we could leave that one out for now, if you add a form to select the data that should be plotted by the graph.

**Comment by Bachibouzouk** [comment]
Ok. This is because in order to test if things works server side I wanted to generate a view with a plotly graph and see it

**Comment by MGlauer** [comment]
I am not sure I understand. The only task done on the server side is collecting a part of the data that was requested via the API. The actual plotting is done in JS on the client side. So, I don't see what you want to plot on the server-side.

**Comment by Bachibouzouk** [comment]
> I am not sure I understand. The only task done on the server side is collecting a part of the data that was requested via the API. The actual plotting is done in JS on the client side. So, I don't see what you want to plot on the server-side.

I see that I lack a precise vocabulary to express what want to do. Also, I probably did not understand what you meant by 
> Tables and maps draw their data through the API and so should graphs.

I would like to know what you would consider a solution to #491 (for my understanding it would involve changing something in the function `plot_view` in the `dataedit/static/dataedit/views.js`)

**Comment by Bachibouzouk** [comment]
@MGlauer - if I want to test this branch, I don't have the same data table as the one you showed in [here](https://github.com/OpenEnergyPlatform/oeplatform/issues/486#issuecomment-557598007). Would it be possible to design a test dataset that one could use for test purposes?

**Comment by MGlauer** [comment]
On this branch all the functionality has been moved to the `backend.js` as `view.js` mostly relied on recline and did not support the new libraries. Therefore, the new functionality should to to `backend.js`. It should retrieve the data via AJAX and create a plot in a div in `dataedit/templates/dataedit/dataview.html`. But let's move more detailed discussions to #491

**Comment by Bachibouzouk** [comment]
This is what I see when I try the selection rules :
Screenshot from 2019-11-27 16-08-49

are they implemented yet?

**Comment by MGlauer** [comment]
Interesting. It looks like the table is detached from it's header. I am not sure how this can happen but will look into this

**Comment by MGlauer** [comment]
Also some of the filters do not work yet. (Because the API does not support them) I am working on it.

**Comment by Bachibouzouk** [comment]
I build the table using https://github.com/OpenEnergyPlatform/examples/blob/master/api/OEP_API_tutorial_part1.ipynb

File: dataedit/static/dataedit/backend.js [hunk 1]
```diff
@@ -227,6 +235,7 @@ function load_table (schema, table, csrftoken) {
         }
         if(table_info.geo_columns.length>0){
             map = L.map('map');
+            debugger;
```
Comment (Bachibouzouk) [inline dataedit/static/dataedit/backend.js]: Belongs to the commit?

Comment (MGlauer) [reply dataedit/static/dataedit/backend.js]: Oh, no...

**Pull request merged**